.github/workflows/ci.yml

```diff
@@ -193,9 +193,16 @@ jobs:
               (cd $dir && terraform plan -input=false)
           done
 
+  provion-self-hosted-runner:
+    name: provion-self-hosted-runner
+    uses: ./.github/workflows/deploy-self-hosted-runner.yml
+    secrets: inherit
+
   k3d_test:
     name: k3d_test
-    runs-on: ubuntu-latest
+    runs-on: self-hosted
```

Comment:
Cool. I'm curious though, how would GitHub know where/how you have self-hosted it?

Reply:
- The Azure storage is used as a lockdown mechanism to avoid the scenario where we have multiple runs at the same time in the VM and one finishes before the others, what the lock mechanism does is create a file with the run ID within the storage whenever a new run is called and before destroying the VM it first deletes the file that has the same run id and then does a check if there are any other files left, if none is left it allows the destroy if there is even one file left it means some run is still in progress and only last run would be allowed to destroy the VM
- GH detects that because the VM is configured and connected to the GH action server and it has a Heartbeats test that checks that the self-hosted runner is still running and lunch the runs within that connected runner (the self-hosted is the default name to any runner so if there is multiple it will run on free one otherwise we can give specific names to each runner)